### ISSUE TITLE
ESLint linter doesn't honor .eslintignore file

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode-eslint#6]
I would like to know how about how to use  "eslint.options" ? I am really not having much luck with getting .eslintignore files to work. 

I have tried placing it in the root directory, this doesn't appear to work.

I have tried adding the .eslintignore file using  "eslint.options" :
    "eslint.options": {
        "ignorePath": ".eslintignore" 
    }

And also with an absolute path:
    "eslint.options": {
        "ignorePath": "/path/to/this/file/.eslintignore" 
    }

I have also tried adding the patterns with the  "ignorePattern"  option:
    "eslint.options": {
        "ignorePattern": "somefile.js"
    }

All without luck! All on OS X. 

I am making the assumption here that this is not a bug and rather its my miss-understanding of the implementation...

**Comment by unional** [comment]
Same in Windows environment

**Comment by Fotiman** [comment]
I would like to see this bug fixed as well. Not being able to ignore 3rd party libraries, for example, is problematic.

**Comment by ArianeLu** [comment]
Same issue here, too many error and warning message cause vscode can't work anymore (with node_modules in project folder.

**Comment by dbaeumer** [comment]
I tried to reproduce this and wasn't able to do so using a simple setup. 

See the two screen shots. 

![capture](https://cloud.githubusercontent.com/assets/1931590/18514030/384eea46-7a90-11e6-82aa-21e2bacf80b2.png)

![capture](https://cloud.githubusercontent.com/assets/1931590/18514059/502b5708-7a90-11e6-920a-0053574cfa1f.png)

Can some of you share your setup so that I can investigate why this isn't working for you.

**Comment by ArianeLu** [comment]
![sp160920_143425](https://cloud.githubusercontent.com/assets/4962408/18659822/a7abdb98-7f3f-11e6-9034-e3e1f9812f5c.png)

@dbaeumer I have add node_modules to .eslintignore, but it still cause vscode stop response after install HTML CSS Class Completion and open a folder contain node_modules with a lot of npm package.

Or it can't match folder, only file?

**Comment by dbaeumer** [comment]
@ArianeLu this looks like a different issue unrelated to eslint. Could you share your workspace so that we can investigate this?

**Comment by dbaeumer** [comment]
I am closing this issue since I can't reproduce and @ArianeLu issue is different.

**Comment by sonnn** [comment]
I have the same problem. 
could you try with .eslintignore with  **/*.js ? mac environment.

**Comment by dbaeumer** [comment]
@sonnn do you have a workspace I can reproduce I could clone.

**Comment by unional** [comment]
Just encountered this still.... in windows and using `.eslintignore`

```sh
// .eslintignore
dbin/*
```
still linting `dbin/x.js`

**Comment by lrowe** [comment]
With or without an .eslintignore, I'm seeing eslint retporting problems for open javascript files within my node_modules (which should be ignored by default.) This is on a Mac with VS Code 1.9.0.

**Comment by dbaeumer** [comment]
@Irowe the linting of files inside the node_modules folder was caused by a bug I introduced in 1.2.3. If fix this in 1.2.4.

**Comment by AlexEvanczuk** [comment]
Hi @dbaeumer has this been fixed?  I'm still seeing the same issue:

`.eslintignore`
```
node_modules/
coverage/
```

eslint results:
```
/home/u6524/src/repo/turnip/node_modules/eslint/bin/eslint.js:16
const useStdIn = (process.argv.indexOf("--stdin") > -1),
^^^^^
...
```

**Comment by cdierkens** [comment]
@dbaeumer Open files are still being linted in VS Code 1.12.0 and ESLint extension 1.2.8

**Comment by dbaeumer** [comment]
@cdierkens can you provide a workspace to reproduce. I can't not using my setup:

![cast](https://cloud.githubusercontent.com/assets/1931590/25944142/8d3a81be-3642-11e7-8d0c-86014c66a69a.gif)

**Comment by TimMeade** [comment]
Did anyone find a fix for this on OSX?  Just installed vscode for the first time and the very first project and node_modules has 100's of issues with eslint.    Deal breaker.....

**Comment by ensean** [comment]
@dbaeumer I have created a [repo](https://github.com/ensean/my-web-store.git) to reproduce the issue    
The demo web-store project is combined by the frontend and backend. The ```frontend``` is an webpack project create by vue-cli, the ```backend``` is an django project.  

I have tried to config the ignorePath param in the workspace's settings.json as follows   
```javascript
{
  "eslint.options": {
    "ignorePath": "${workspaceRoot}/frontend/.eslintignore"
  }
}
```
But it seems that eslint-plugin can not read the eslintignore file?

![image](https://user-images.githubusercontent.com/12255172/32600753-3ce3aa58-c506-11e7-97c3-8f45fa226281.png)
